scripts/data_prep/convert_delta_to_json.py

```diff
@@ -0,0 +1,451 @@
+    # Iterate over the response
+
+    result = []
+    row_count = 0
+    is_overflow = False
+
+    for response in iterator:
+        if response.HasField('extension') and response.extension.Is(
+                cloud_pb2.CloudResultBatch.DESCRIPTOR):
+            batch = cloud_pb2.CloudResultBatch()
+            assert response.extension.Is(cloud_pb2.CloudResultBatch.DESCRIPTOR)
+            response.extension.Unpack(batch)
+            result += [
+                Result(b.url, b.row_count, b.compressed_size,
+                       b.uncompressed_size) for b in batch.results
+            ]
+            row_count += sum(result.row_count for result in batch.results)
+            is_overflow |= batch.truncated
+    return result, row_count, is_overflow
+
+
+SparkConnectClient.to_cf = to_cf  # pyright: ignore
+
+
+def collect_as_cf(self: DataFrame,
+                  type: str = 'json') -> Tuple[List[Result], int, bool]:
+    query = self._plan.to_proto(self._session.client)  # pyright: ignore
+    return self._session.client.to_cf(query, type)  # pyright: ignore
+
+
+DataFrame.collect_cf = collect_as_cf  # pyright: ignore
+
+
+def iterative_combine_jsons(json_directory: str, output_file: str) -> None:
+    """Combine jsonl files in json_directory into one big jsonl file.
+
+    Args:
+        json_directory(str): directory containing the JSON files
+        output_file(str): output JSONL file
+    """
+    json_files = [f for f in os.listdir(json_directory) if f.endswith('.jsonl')]
+    with open(output_file, 'w') as outfile:
+        for file_name in json_files:
+            with open(os.path.join(json_directory, file_name), 'r') as infile:
+                for line in infile:
+                    outfile.write(line)
+    print('JSON files have been combined into a JSONL file.')
+
+
+def run_query(
+    q: str,
+    method: str,
+    cursor: Optional[Cursor] = None,
+    spark: Optional[SparkSession] = None,
+    collect: bool = True
+) -> Optional[Union[List[Row], DataFrame, SparkDataFrame]]:
+    """Run SQL query via databricks-connect or databricks-sql.
+
+    Args:
+        q (str): sql query
+        method (str): select from dbsql and dbconnect
+        cursor (Cursor): connection.cursor
+        spark (SparkSession): spark session
+        collect (bool): whether to get the underlying data from spark dataframe
+    """
+    if method not in ['dbsql', 'dbconnect']:
+        raise ValueError(f'Unrecognized method: {method}')
+
+    if method == 'dbsql':
+        if cursor is None:
+            raise ValueError(f'cursor cannot be None if using method dbsql')
+        cursor.execute(q)
+        if collect:
+            return cursor.fetchall()
+
+    elif method == 'dbconnect':
+        if spark == None:
+            raise ValueError(f'sparkSession is required for dbconnect')
+        df = spark.sql(q)
+        if collect:
+            return df.collect()
+        return df
+
+    return None
```

Comment:
i don't think the return none should be here. this code is expected to never execute, right?

Reply:
im guessing that would also change the return type to not be `Optional`

Reply:
yes, it should never reach ``None``. Fixed.